pennylane_lightning_gpu/src/util/DataBuffer.hpp

```diff
@@ -125,6 +128,40 @@ template <class GPUDataT, class DevTagT = int> class DataBuffer {
             cudaMemset(gpu_buffer_, 0, length_ * sizeof(GPUDataT)));
     }
 
+    /**
+     * @brief Set value to ith element
+     *
+     * @params index Index of the element to be set.
+     * @params value The value passed to the target element.
+     *
+     */
+    template <class HostDataT = GPUDataT>
+    void setIthElement(const HostDataT &value, const size_t index,
+                       bool async = false) {
+        if (!async) {
+            PL_CUDA_IS_SUCCESS(cudaMemcpy(&gpu_buffer_[index], &value,
+                                          sizeof(GPUDataT),
+                                          cudaMemcpyHostToDevice));
+        } else {
+            PL_CUDA_IS_SUCCESS(
+                cudaMemcpyAsync(&gpu_buffer_[index], &value, sizeof(GPUDataT),
+                                cudaMemcpyHostToDevice, getStream()));
+        }
+    }
+
+    /**
+     * @brief Set value to ith element
+     *
+     * @params num_indices Number of elements to be set.
+     * @params value The value pointer passed to the target elements.
+     * @params indices The indices of the target elements.
+     *
+     */
+    template <class DeviceDataT, class index_type>
+    void setElements(index_type &num_indices, DeviceDataT *value,
+                     index_type *indices, size_t thread_per_block,
+                     cudaStream_t stream_id);
```

Comment:
We may want to rethink this design. I think it better to define CUDA kernels external to the DataBuffer class, but have them manipulate the data elements within, rather than explicitly adding kernel-defined methods to the class itself.

I need to better think about this, but keeping dtabuffer as is, and having another GPUKernels class, or something else, that accepts a databuffer and operators on it would be preferred.

Reply:
One question, what's the reason in your mind about it's better to define CUDA kernels external to the `DataBuffer class`.